src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Extensions/PropertyExtensions.cs

```diff
@@ -7,26 +7,30 @@ namespace VirtoCommerce.XDigitalCatalog.Extensions
 {
     public static class PropertyExtensions
     {
-        public static IList<Property> ExpandByValues(this IEnumerable<Property> properties)
-        {
-            var result = properties.SelectMany(property =>
-            {
-                var propertyValues = property.Values.Select(v =>
+        public static IList<Property> ExpandByValues(this IEnumerable<Property> properties, string cultureName)
+            => properties.SelectMany(property => property.Values
+                .GroupBy(propertyValue => propertyValue.Alias)
+                .Select(aliasGroup => cultureName switch
                 {
-                    var clonedProperty = (Property)property.Clone();
-                    clonedProperty.Values = new List<PropertyValue> { v };
-                    return clonedProperty;
-                }).ToList();
+                    string languageCode when languageCode != null
```

Comment:
Can we avoid Query Syntax and use Method syntax for Linq expression? In addition due to many places where we need to use such filtration by language (product descriptions etc) need to create an extension method and use it instead.  see example https://github.com/VirtoCommerce/vc-storefront/blob/master/VirtoCommerce.Storefront.Model/Common/LocalizationExtensions.cs#L58 
PS. pay attention to the fallback logic for choosing the proper localized string in the given example

Reply:
1) Query syntax is more compact and more beautiful for me :) 
2) This code also returns default values for nonlocalized property values, so this code is not a pure filtration method, btw creating an extension method for IGrouping<string, PropertyValue>is redundant;
3) It's not refactoring PR.

If further discussion is needed we can call.